### PR TITLE
Update language based on discussions in July 2023

### PR DESCRIPTION
**Pull request by ftomassetti**
This align the Java implementation to the changes in the M3 introduced lately.
It adds tests to verify that the builtin languages defined here correspond to the ones part of the documentation.

### CODE REVIEW

**Review by enikao — APPROVED** [review]

File: core/src/main/java/io/lionweb/lioncore/java/model/impl/DynamicNode.java [hunk 1]
```diff
@@ -246,7 +246,7 @@ public String toString() {
     String qualifiedName;
     try {
       qualifiedName = concept.qualifiedName();
-    } catch(Throwable t) {
+    } catch (Throwable t) {
```
[the remark this answers is not in the document]
Comment (ftomassetti) [reply core/src/main/java/io/lionweb/lioncore/java/model/impl/DynamicNode.java:249]: Right, changing it to RuntimeException

File: core/src/main/java/io/lionweb/lioncore/java/self/LionCore.java [hunk 2]
```diff
@@ -96,119 +88,97 @@ public static Language getInstance() {
...
 
   private static void checkIDs(M3Node node) {
-    Set<String> clashingKeys = new HashSet<>(Arrays.asList("type", "extends", "name"));
     if (node.getID() == null) {
       if (node instanceof NamespacedEntity) {
         NamespacedEntity namespacedEntity = (NamespacedEntity) node;
-        node.setID(namespacedEntity.qualifiedName().replaceAll("\\.", "_"));
-        if (node instanceof HasKey<?>) {
+        node.setID("-id-" + namespacedEntity.getName().replaceAll("\\.", "_"));
+        if (node instanceof HasKey<?> && ((HasKey<?>) node).getKey() == null) {
```
[the remark this answers is not in the document]
Comment (ftomassetti) [reply core/src/main/java/io/lionweb/lioncore/java/self/LionCore.java:181]: Right, now that we have IKeyed, I will rename HasKey to IKeyed

File: core/src/main/java/io/lionweb/lioncore/java/serialization/JsonSerialization.java [hunk 3]
```diff
@@ -259,13 +261,22 @@ private void serializeNodeProperties(Node node, SerializedNode serializedNode) {
...
+  }
+
   public List<Node> unserializeToNodes(JsonElement jsonElement) {
     SerializedChunk serializationBlock =
         new LowLevelJsonSerialization().unserializeSerializationBlock(jsonElement);
     validateSerializationBlock(serializationBlock);
     return unserializeSerializationBlock(serializationBlock);
   }
 
+  public List<Node> unserializeToNodes(URL url) throws IOException {
```
[the remark this answers is not in the document]
Comment (ftomassetti) [reply core/src/main/java/io/lionweb/lioncore/java/serialization/JsonSerialization.java:275]: I could not find anything stating that URL is deprecated, and I cannot find a way to read URis, so I would keep the URL for now

File: core/src/test/java/io/lionweb/lioncore/java/language/BuiltinIDsTest.java [hunk 4]
```diff
@@ -8,75 +8,61 @@ public class BuiltinIDsTest {
 
   @Test
   public void M3ElementsHasExpectedIDs() {
```
[the remark this answers is not in the document]
Comment (ftomassetti) [reply core/src/test/java/io/lionweb/lioncore/java/language/BuiltinIDsTest.java:10]: Adding the test for that

File: core/src/main/java/io/lionweb/lioncore/java/serialization/JsonSerialization.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (enikao) [reply core/src/main/java/io/lionweb/lioncore/java/serialization/JsonSerialization.java:275]: ok

File: core/src/test/resources/serialization/lioncore.serialization.schema.json [hunk 5]
```diff
@@ -197,32 +197,32 @@
...
-            "LIonCore_M3_Feature_optional": {
+            "LIonCore-M3-Feature-optional": {
               "type": "string"
             },
-            "LIonCore_M3_Feature_derived": {
+            "LIonCore-M3-Feature-derived": {
               "type": "string"
             },
-            "LIonCore_M3_NamespacedEntity_simpleName": {
+            "LIonCore-M3-NamespacedEntity-simpleName": {
```
[the remark this answers is not in the document]
Comment (ftomassetti) [reply core/src/test/resources/serialization/lioncore.serialization.schema.json:217]: Right, this file is not actually used. I am going to remove it

**Comment by ftomassetti** [comment]
Thanks for the review @enikao !

File: core/src/main/java/io/lionweb/lioncore/java/language/NamespacedEntity.java [hunk 6]
```diff
@@ -6,8 +6,7 @@
  * <p>A Concept Invoice, contained in a Language com.foo.Accounting. Therefore, Invoice will have
  * the qualifiedName com.foo.Accounting.Invoice.
  */
-public interface NamespacedEntity {
-  // TODO add ID, once details are clearer
+public interface NamespacedEntity extends INamed {
```
Comment (enikao) [inline core/src/main/java/io/lionweb/lioncore/java/language/NamespacedEntity.java:9]: This is not part of "official" M3, but as discussed an implementation can have other interfaces. Just to make sure this is a conscious decision.

**Pull request merged**